File: source/322.零钱兑换.cpp
/*
 * @lc app=leetcode.cn id=322 lang=cpp
 *
 * [322] 零钱兑换
 */
#include "cpp_includes.h"
// @lc code=start
class Solution {
 public:
  int coinChange(vector<int>& coins, int amount) {
    vector f(amount + 1, amount + 1);
    f[0] = 0;
    for (int i = 1; i <= amount; i++) {
      for (auto c : coins) {
        if (i >= c) {
          f[i] = min(f[i], f[i - c] + 1);
        }
      }
    }

    return f[amount] == amount + 1 ? -1 : f[amount];
  }
};
// @lc code=end

File: source/322.零钱兑换.py
#
# @lc app=leetcode.cn id=322 lang=python3
#
# [322] 零钱兑换
#

# @lc code=start
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        f = [amount+1]*(amount+1)
        f[0] = 0
        for i in range(0, amount+1):
            for coin in coins:
                if i >= coin:
                    f[i] = min(f[i], f[i-coin]+1)
        
        return f[amount] if f[amount] != amount+1 else -1
# @lc code=end

